android-build-tools/src/readelf.rs

```diff
@@ -0,0 +1,113 @@
+use crate::apk::UnalignedApk;
+use crate::error::NdkError;
+use crate::target::Target;
+use std::collections::HashSet;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+pub trait Readelf {
```

Comment:
Are the reason to have additional trait for that?
I think that enough add extra impl for `UnalignedApk` itself.

```rust
impl<'a> UnalignedApk<'a> {
    fn add_lib_recursively ...
}
```